src/ol_infrastructure/applications/superset/__main__.py

```diff
@@ -0,0 +1,742 @@
+    security_groups=[
+        superset_security_group.id,
+        consul_security_groups["consul_agent"],
+    ],
+    tags=aws_config.merged_tags({"Name": f"superset-worker-{stack_info.env_suffix}"}),
+    tag_specifications=superset_worker_tag_specs,
+    user_data=consul_datacenter.apply(
+        lambda consul_dc: base64.b64encode(
+            "#cloud-config\n{}".format(
+                yaml.dump(
+                    {
+                        "write_files": [
+                            {
+                                "path": "/etc/consul.d/02-autojoin.json",
+                                "content": json.dumps(
+                                    {
+                                        "retry_join": [
+                                            "provider=aws tag_key=consul_env "
+                                            f"tag_value={consul_dc}"
+                                        ],
+                                        "datacenter": consul_dc,
+                                    }
+                                ),
+                                "owner": "consul:consul",
+                            },
+                            {
+                                "path": "/etc/default/vector",
+                                "content": textwrap.dedent(
+                                    f"""\
+                            ENVIRONMENT={consul_dc}
+                            APPLICATION=superset
+                            SERVICE=data-platform
+                            VECTOR_CONFIG_DIR=/etc/vector/
+                            AWS_REGION={aws_config.region}
+                            GRAFANA_CLOUD_API_KEY={grafana_credentials['api_key']}
+                            GRAFANA_CLOUD_PROMETHEUS_API_USER={grafana_credentials['prometheus_user_id']}
+                            GRAFANA_CLOUD_LOKI_API_USER={grafana_credentials['loki_user_id']}
+                            """
+                                ),
+                                "owner": "root:root",
+                            },
+                            {
+                                "path": "/etc/default/superset",
+                                "content": f"DOMAIN={superset_config.get('domain')}",
+                            },
+                            {
+                                "path": "/etc/default/docker-compose",
+                                "content": "COMPOSE_PROFILES=worker",
+                            },
+                            {
+                                "path": "/etc/profile",
+                                "content": "export COMPOSE_PROFILES=worker",
+                                "append": True,
+                            },
+                            {
+                                "path": "/etc/docker/compose/.env",
+                                "content": f"DOMAIN={superset_domain}\nVAULT_ADDR=https://vault-{stack_info.env_suffix}.odl.mit.edu\n",
```

Comment:
I believe it is this reference to DOMAIN that gets interpolated into the traefik labels in the docker-compose.yaml file.

Reply:
Also, do the workers need this at all?

Reply:
Yes, I believe the workers need to know the domain to ensure that e.g. emails have proper values.